contracts/cw2981-royalties/src/msg.rs

```diff
@@ -0,0 +1,138 @@
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// Total number of tokens issued
+    NumTokens {},
+
+    /// With MetaData Extension.
+    /// Returns top-level metadata about the contract: `ContractInfoResponse`
+    ContractInfo {},
+    /// With MetaData Extension.
+    /// Returns metadata about one particular token, based on *ERC721 Metadata JSON Schema*
+    /// but directly from the contract: `NftInfoResponse`
+    NftInfo {
+        token_id: String,
+    },
+    /// With MetaData Extension.
+    /// Returns the result of both `NftInfo` and `OwnerOf` as one query as an optimization
+    /// for clients: `AllNftInfo`
+    AllNftInfo {
+        token_id: String,
+        /// unset or false will filter out expired approvals, you must set to true to see them
+        include_expired: Option<bool>,
+    },
+
+    /// With Enumerable extension.
+    /// Returns all tokens owned by the given address, [] if unset.
+    /// Return type: TokensResponse.
+    Tokens {
+        owner: String,
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// With Enumerable extension.
+    /// Requires pagination. Lists all token_ids controlled by the contract.
+    /// Return type: TokensResponse.
+    AllTokens {
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+}
+
+impl From<Cw2981QueryMsg> for CW721QueryMsg {
+    fn from(msg: Cw2981QueryMsg) -> CW721QueryMsg {
+        match msg {
+            Cw2981QueryMsg::OwnerOf {
+                token_id,
+                include_expired,
+            } => CW721QueryMsg::OwnerOf {
+                token_id,
+                include_expired,
+            },
+            Cw2981QueryMsg::ApprovedForAll {
+                owner,
+                include_expired,
+                start_after,
+                limit,
+            } => CW721QueryMsg::ApprovedForAll {
+                owner,
+                include_expired,
+                start_after,
+                limit,
+            },
+            Cw2981QueryMsg::NumTokens {} => CW721QueryMsg::NumTokens {},
+            Cw2981QueryMsg::ContractInfo {} => CW721QueryMsg::ContractInfo {},
+            Cw2981QueryMsg::NftInfo { token_id } => CW721QueryMsg::NftInfo { token_id },
+            Cw2981QueryMsg::AllNftInfo {
+                token_id,
+                include_expired,
+            } => CW721QueryMsg::AllNftInfo {
+                token_id,
+                include_expired,
+            },
+            Cw2981QueryMsg::Tokens {
+                owner,
+                start_after,
+                limit,
+            } => CW721QueryMsg::Tokens {
+                owner,
+                start_after,
+                limit,
+            },
+            Cw2981QueryMsg::AllTokens { start_after, limit } => {
+                CW721QueryMsg::AllTokens { start_after, limit }
+            }
+            _ => panic!("cannot covert {:?} to CW721QueryMsg", msg),
```

Comment:
Don't think we want a panic here...

Reply:
You need it to match all paths, even though it should never be hit because you extend the outer enum.